src/test/java/org/apache/commons/mail/resolver/DataSourceCompositeResolverTest.java

```diff
@@ -58,12 +60,13 @@ public void testResolvingFilesLenient() throws Exception
         assertTrue(toByteArray(dataSourceResolver.resolve("/contentTypeTest.gif")).length > 0);
     }
 
-    @Test(expected = IOException.class)
-    public void testResolvingFilesNonLenient() throws Exception
+    @Test
+    public void testResolvingFilesNonLenient()
     {
         final DataSourceResolver dataSourceResolver = new DataSourceCompositeResolver(dataSourceResolversMissing, false);
 
-        dataSourceResolver.resolve("./image/does-not-exist.gif");
+        final Executable testMethod = () -> dataSourceResolver.resolve("./image/does-not-exist.gif");
```

Comment:
Hello @nhojpatrick 
Thank you for your PR.
As with the other PRs, the local variable is superfluous. 
Please rebase on master to fix the spotbugs build on Java 11 and 17.